### PR TITLE
chore: add CCTPDepositOnly FF

### PR DESCRIPTION
**Pull request by yogurtandjam**
with CCTPDepositOnly enabled
Screenshot 2024-05-15 at 11 41 25 AM
Screenshot 2024-05-15 at 11 41 36 AM

with CCTPDepositOnlyDisabled
Screenshot 2024-05-15 at 11 43 05 AM
Screenshot 2024-05-15 at 11 43 13 AM

### CODE REVIEW

File: src/views/forms/AccountManagementForms/SourceSelectMenu.tsx [hunk 1]
```diff
@@ -85,10 +85,14 @@ export const SourceSelectMenu = ({
...
       ),
     }))
-    .filter(
-      (chain) =>
-        type === TransferType.deposit || !!cctpTokensByChainId[chain.value] || !CCTPWithdrawalOnly
-    )
+    .filter((chain) => {
+      if (type === TransferType.deposit) {
+        // if deposit, return only CCTP tokens UNLESS CCTPDepositOnly is not enabled
+        return !!cctpTokensByChainId[chain.value] || !CCTPDepositOnly;
```
Comment (yogurtandjam) [inline src/views/forms/AccountManagementForms/SourceSelectMenu.tsx:91]: [note] we can condense this all into a 1 liner but i felt like expanding this and adding comments helps make this more clear and extendable for any future FF's or features

File: src/hooks/useEnvFeatures.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export interface EnvironmentFeatures {
   reduceOnlySupported: boolean;
   withdrawalSafetyEnabled: boolean;
   CCTPWithdrawalOnly: boolean;
+  CCTPDepositOnly?: boolean;
```
Comment (moo-onthelawn) [inline src/hooks/useEnvFeatures.ts:11]: should this be optional?

**Review by moo-onthelawn — COMMENTED** [review]
hm are we missing the changes to env.json (that set this to be true or false?)? Or are you using the optional feature flag to basically default it to false? (It feels slightly clearer to explicitly set everything to false, but if that makes it harder for DOS this could be ok 🤔 )

**Comment by yogurtandjam** [comment]
> hm are we missing the changes to env.json (that set this to be true or false?)? Or are you using the optional feature flag to basically default it to false? (It feels slightly clearer to explicitly set everything to false, but if that makes it harder for DOS this could be ok 🤔 )

so my general philosophy is that FF's/env vars should be optional since the application can't guarantee the existence of externally injected variables. but i think to make sure that DOS doesn't have trouble missing env file changes it makes sense to make it required

File: src/views/forms/AccountManagementForms/SourceSelectMenu.tsx [hunk 3]
```diff
@@ -85,10 +85,14 @@ export const SourceSelectMenu = ({
         </Styled.Text>
       ),
     }))
-    .filter(
-      (chain) =>
-        type === TransferType.deposit || !!cctpTokensByChainId[chain.value] || !CCTPWithdrawalOnly
-    )
+    .filter((chain) => {
```
Comment (moo-onthelawn) [inline src/views/forms/AccountManagementForms/SourceSelectMenu.tsx:88]: nit would this be equivalent to: `if (type === TransferType.deposit && CCTPDepositOnly) || (type === TransferType.widthdrawal && CCTPWithdrawalOnly) return  !!cctpTokensByChainId[chain.value] else true`

Comment (yogurtandjam) [reply src/views/forms/AccountManagementForms/SourceSelectMenu.tsx:88]: is that easier for you to understand? i [just tried to write it in the way that felt the clearest for me to read](https://github.com/dydxprotocol/v4-web/pull/540/files#r1601887869) - if it's easier for other people to understand it that way i'll rewrite it!

Comment (moo-onthelawn) [reply src/views/forms/AccountManagementForms/SourceSelectMenu.tsx:88]: eh I don't wanna block on this nit; also don't think it's a big deal, up to you!

**Review by moo-onthelawn — APPROVED** [review]

File: src/views/forms/AccountManagementForms/SourceSelectMenu.tsx (hunk 3 above)
Comment (yogurtandjam) [reply src/views/forms/AccountManagementForms/SourceSelectMenu.tsx:88]: synced offline and updated to a mix of both @moo-onthelawn's suggestion and my original!

**Pull request merged**